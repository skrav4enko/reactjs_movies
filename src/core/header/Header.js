import React from 'react';
import './Header.scss';
import Logo from '../../shared/logo/Logo';

import SearchIcon from '../../assets/icons/search-24px.svg';

function Header(props) {
  return (
    <header className="header">
      <div className="container">
        <Logo />
        <span className="header__icon">
          <SearchIcon />
        </span>
      </div>
    </header>
  );
}

export default Header;
