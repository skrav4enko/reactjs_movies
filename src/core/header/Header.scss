.header {
  background-color: transparent;
  position: absolute;
  top: 0;
  left: 0;
  height: 70px;
  width: 100%;

  .container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 100%;
    max-width: 1200px;
    margin: 0 auto;
  }

  &__icon {
    fill: #F65261;
    cursor: pointer;
    height: 32px;
    width: 32px;

    svg {
      display: block;
      height: 100%;
      width: 100%;
    }
  }
}
