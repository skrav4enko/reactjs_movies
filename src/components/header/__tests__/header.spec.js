/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Header from '..';

describe('Header', () => {
  it('should match snapshot', () => {
    const navigateTo = jest.fn();
    const wrapper = shallow(<Header navigateTo={navigateTo} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call navigateTo', () => {
    const navigateTo = jest.fn();
    const component = shallow(<Header navigateTo={navigateTo} />);
    const icon = component.find('.header__icon').simulate('click');
    expect(navigateTo).toHaveBeenCalled();
  });
});

