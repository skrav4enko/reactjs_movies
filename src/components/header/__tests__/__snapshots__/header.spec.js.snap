// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <Logo />
    <NavLink
      className="header__icon"
      exact={true}
      to="/"
    >
      <search-24px.svg />
    </NavLink>
  </div>
</header>
`;
